### DISCUSSION TITLE
[push_to_git_remote] Support git option `push-option`

### DISCUSSION THREAD

**Discussion by ShivaHuang** [fastlane/fastlane#17719] [Ideas] [answered]
I use fastlane to work with gitlab-ci and encountered the following situation.

In my CI configuration, a tag will be pushed to repo if the pipeline success.
However, the push triggered by fastlane will trigger another pipeline. The solution I found is use git option `--push-option=ci.skip` to tell gitlab don't trigger CI pipeline.

It works, however, `push_to_git_remote` supports only limited git options and `--push-option` is not one of them.
So I have to write the push actions in CI configuration but not in fastlane.

Is it possible to add `--push-option` to `push_to_git_remote`?
Or a more general options argument is preferred?

Ref:
- [1] https://docs.gitlab.com/ee/ci/yaml/#skip-pipeline
- [2] https://git-scm.com/docs/git-push#Documentation/git-push.txt---push-optionltoptiongt

**Comment by rogerluan** [comment, marked as answer]
Hello @ShivaHuang 👋 

That's a great suggestion! I didn't know of this specific `git` feature (probably because I never used a self-hosted git server 😊). 

I've implemented that feature here: https://github.com/fastlane/fastlane/pull/17791

Be my guest to test it if you want to! Testing steps are included in the PR description 🤗 

Hope that helps you out!

**Reply by ShivaHuang** [reply]
Hi @rogerluan 

I've tested it and it works like a charm 🎉😍

Thank you for your great job!

**Reply by rogerluan** [reply]
Glad I could help @ShivaHuang ! 💪
